### PR TITLE
new endpoints and tweaks

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=3284018&stafilter=NotComplete&fileview=grid

Signed-off-by: romanetar <roman_ag@hotmail.com>

### CODE REVIEW

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitDocumentsApiController.php [hunk 1]
```diff
@@ -343,4 +344,41 @@ public function removeEventType($summit_id, $document_id, $event_type_id){
...
+     * @param $summit_id
+     * @param $document_id
+     * @return \Illuminate\Http\JsonResponse|mixed
+     */
+    public function removeFile($summit_id, $document_id){
+        return $this->processRequest(function () use ($summit_id, $document_id) {
+            $summit = SummitFinderStrategyFactory::build($this->summit_repository, $this->resource_server_context)->find($summit_id);
+            if (is_null($summit)) return $this->error404();
+
+            $document = $this->service->removeFileFromSummitDocument($summit, $document_id);
```
Comment (smarcet) [inline app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitDocumentsApiController.php:379]: document_id should be parsed to intval(document_id)

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitDocumentsApiController.php [hunk 2]
```diff
@@ -343,4 +344,41 @@ public function removeEventType($summit_id, $document_id, $event_type_id){
...
+
+            return $this->deleted(SerializerRegistry::getInstance()->getSerializer($document)->serialize());
```
Comment (smarcet) [inline app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitDocumentsApiController.php:381]: add missing serializer params

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitDocumentsApiController.php [hunk 3]
```diff
@@ -343,4 +344,41 @@ public function removeEventType($summit_id, $document_id, $event_type_id){
...
+            if (is_null($summit)) return $this->error404();
+
+            $file = $request->file('file');
+            if (is_null($file)) {
+                return $this->error412(array('file param not set!'));
+            }
+
+            $document = $this->service->addFile2SummitDocument($summit, $document_id, $file);
+
+            return $this->updated(SerializerRegistry::getInstance()->getSerializer($document)->serialize());
```
Comment (smarcet) [inline app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitDocumentsApiController.php:365]: add missing serializer params

    SerializerUtils::getExpand(),
    SerializerUtils::getFields(),
    SerializerUtils::getRelations()

File: routes/api_v1.php [hunk 4]
```diff
@@ -1043,6 +1043,12 @@
                 Route::get('', 'OAuth2SummitDocumentsApiController@get');
                 Route::put('', ['middleware' => 'auth.user', 'uses' => 'OAuth2SummitDocumentsApiController@update']);
                 Route::delete('', ['middleware' => 'auth.user', 'uses' => 'OAuth2SummitDocumentsApiController@delete']);
+
+                Route::group(['prefix' => 'file'], function () {
```
Comment (smarcet) [inline routes/api_v1.php:1047]: add where restriction on document id to be only number

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review comments

File: app/Services/Model/Imp/SummitDocumentService.php [hunk 5]
```diff
@@ -272,4 +255,59 @@ public function removeEventTypeFromSummitDocument(Summit $summit, int $document_
...
+
+            $attachment = $this->file_uploader->build
+            (
+                $file,
+                sprintf('summits/%s/documents', $summit->getId()),
+                false
+            );
+
+            //clear web_link as it's no longer required when a file is added
+            $document->setWebLink('');
```
Comment (smarcet) [inline app/Services/Model/Imp/SummitDocumentService.php:281]: can we create a clearWebLink method?

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitDocumentsApiController.php (hunk 2 above)
Comment (romanetar) [reply app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitDocumentsApiController.php:381]: done

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitDocumentsApiController.php (hunk 1 above)
Comment (romanetar) [reply app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitDocumentsApiController.php:379]: done

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitDocumentsApiController.php [hunk 6]
```diff
@@ -343,4 +344,41 @@ public function removeEventType($summit_id, $document_id, $event_type_id){
...
+        return $this->processRequest(function () use ($request, $summit_id, $document_id) {
+            $summit = SummitFinderStrategyFactory::build($this->summit_repository, $this->resource_server_context)->find($summit_id);
+            if (is_null($summit)) return $this->error404();
+
+            $file = $request->file('file');
+            if (is_null($file)) {
+                return $this->error412(array('file param not set!'));
+            }
+
+            $document = $this->service->addFile2SummitDocument($summit, $document_id, $file);
```
[the remark this answers is not in the document]
Comment (romanetar) [reply app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitDocumentsApiController.php:363]: done

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitDocumentsApiController.php [hunk 7]
```diff
@@ -343,4 +345,49 @@ public function removeEventType($summit_id, $document_id, $event_type_id){
...
+     * @return \Illuminate\Http\JsonResponse|mixed
+     */
+    public function removeFile($summit_id, $document_id){
+        return $this->processRequest(function () use ($summit_id, $document_id) {
+            $summit = SummitFinderStrategyFactory::build($this->summit_repository, $this->resource_server_context)->find($summit_id);
+            if (is_null($summit)) return $this->error404();
+
+            $document = $this->service->removeFileFromSummitDocument($summit, intval($document_id));
+
+            return $this->deleted(SerializerRegistry::getInstance()->getSerializer($document)->serialize(
```
Comment (smarcet) [inline app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitDocumentsApiController.php:386]: delete dont need to return anything, generally we return 204 without content
check another similar places

**Review by smarcet — APPROVED** [review]
LGTM

**Pull request merged**